internal/service/currency-service.go

```diff
@@ -16,44 +16,50 @@ import (
 	"genesis-currency-api/pkg/util/date"
 )
 
-type CurrencyService struct {
+type CurrencyService interface {
+	GetCurrencyInfo() dto.CurrencyInfoDTO
+	GetCurrencyRate() dto.CurrencyResponseDto
+	UpdateCurrencyRates() error
+}
```

Comment:
Moved interfaces closer to their clients. Since some services are used among different packages, a decision was made to create a separate folder for interfaces in ./pkg. 
![image](https://github.com/GenesisEducationKyiv/software-engineering-school-4-0-AlwaysSayNo/assets/98183720/c48e6430-0e63-4a21-89af-9cffbe648eeb)

Reply:
In Go, you don`t need a separate package for interfaces.